client/network/src/behaviour.rs

```diff
@@ -86,6 +87,11 @@ impl<B: BlockT, H: ExHashT> Behaviour<B, H> {
 		}
 	}
 
+	/// Enable discovery for the given protocol.
+	pub fn add_discovery(&mut self, p: ProtocolId) {
```

Comment:
Since it doesn't make sense to add support for a protocol at runtime, I would instead pass a list of protocols as part of the `new` function.

Reply:
Is it conceivable to run discovery only with MDNS and bootnodes?

Reply:
In practice it doesn't really make sense, but I also don't see a reason to forbid that use case.